addons/account/static/tests/section_and_note.test.js

```diff
@@ -1239,3 +1241,298 @@ test("swap sections and subsections", async () => {
+        message: "Sections should always show hide composition button"
+    });
+    expect(".o-dropdown-item:contains(Hide Prices)").toHaveCount(1, {
+        message: "Sections should always show hide prices button"
+    });
+
+    await contains(".o_data_row:contains(sec1-sub1) .o_list_section_options button").click();
+    expect(".o-dropdown-item:contains(Hide Composition)").toHaveClass("disabled", {
+        message: "Subsection under hidden section should have disabled hide composition button"
+    });
+    expect(".o-dropdown-item:contains(Hide Prices)").toHaveClass("disabled", {
+        message: "Subsection under hidden section should have disabled Hide Prices button"
+    });
+
+    expect(".o_data_row:contains(sec1-r1)").toHaveClass("text-muted", {
+        message: "Line under hidden section should be muted"
+    });
+    expect(".o_data_row:contains(sec1-sub1)").toHaveClass("text-muted", {
+        message: "Subsection under hidden section should be muted"
+    });
+    expect(".o_data_row:contains(sec1-sub1-r1)").toHaveClass("text-muted", {
+        message: "Line under subsection(which is under hidden section) should be muted"
+    });
+
+    expect(".o_data_row:contains(sec2-r1) > td[name='aggregated_field']").toHaveClass("text-muted", {
+        message: "Aggregated field column of Line under hidden prices section should be muted"
+    });
+    expect(".o_data_row:contains(sec2-sub1) > td[name='aggregated_field']").toHaveClass("text-muted", {
+        message: "Aggregated field column of Subsection under hidden prices section should be muted"
+    });
+    expect(".o_data_row:contains(sec2-sub1-r1) > td[name='aggregated_field']").toHaveClass("text-muted", {
+        message: "Aggregated field column of Line under subsection(which is under hidden prices section) should be muted"
+    });
+
+    expect(".o_data_row:contains(sec3-sub1-r1) > td[name='aggregated_field']").toHaveClass("text-muted", {
+        message: "Aggregated field column of Line under hidden prices subsection should be muted"
+    });
+    expect(".o_data_row:contains(sec3-sub2-r1)").toHaveClass("text-muted", {
+        message: "Line under hidden subsection should be muted"
+    });
+})
+
+test("check collapse_ fields' duplicating logic", async () => {
+    InvoiceLine._records[2].collapse_composition = true;
+    InvoiceLine._records[2].collapse_prices = true;
+    InvoiceLine._records[5].collapse_composition = true;
+    InvoiceLine._records[5].collapse_prices = true;
+    InvoiceLine._records[8].collapse_composition = true;
+    InvoiceLine._records[8].collapse_prices = true;
+
+    onRpc("web_save", ({ args }) => {
+        expect.step("web_save");
+
+        const [, { invoice_line_ids }] = args;
+
+        // Filter section/subsection records created via (0, 0, values)
+        const createdSectionRecords = invoice_line_ids.filter(
+            ([cmd, , values]) => cmd === 0 && values?.display_type
+        );
```

Comment:
> ([cmd, , values]) => cmd === 0 && values?.display_type

This will incorrectly match notes, because they also have display_type.

```suggestion
       const createdSectionRecords = invoice_line_ids.filter(([cmd, , values]) =>
            cmd === 0 &&
            values.display_type === ("line_section" || "line_subsection")
        );
```

This prevents future regressions if someone later adds display_type="line_note"